File: demos/ios/Demos/Demos/views/BaseBarChart.m
#import "BaseBarChart.h"

@implementation BaseBarChart

- (void)chartRender {
    NSString *jsonPath = [[NSBundle mainBundle] pathForResource:@"Res/mockData_baseInterval" ofType:@"json"];
    NSString *jsonData = [NSString stringWithContentsOfFile:jsonPath encoding:NSUTF8StringEncoding error:nil];
    self.chart.canvas(self.canvasView).padding(20, 10, 20, 10.f);
    self.chart.source([F2Utils toJsonArray:jsonData]);
    self.chart.axis(@"year", @{@"grid": @(NO)});
    self.chart.axis(@"sales", @{@"grid": @{@"type": @"dash", @"lineWidth": @(2), @"dash": @[@(6), @(2)]}});
    self.chart.scale(@"year", @{@"range": @[@(0.1), @(0.9)]});
    self.chart.scale(@"sales", @{@"nice": @(YES)});
    self.chart.interval().position(@"year*sales").style(@{
        @"rounding": @[@(10), @(10), @(0), @(0)] // [tl, tr, bl, br]
    });
    self.chart.tooltip(@{});
    self.chart.animate(YES);
    self.chart.render();
}

@end

File: demos/ios/Demos-Swift/Demos-Swift/Views/BaseBarChart.swift
//
//  BaseInterval2UIView.swift
//  Demos-Swift
//
//  Created by weiqing.twq on 2022/6/1.
//

import Foundation
import F2

class BaseBarChart: BaseLineChart {
    
    override func chartRender() {
        guard let jsonPath = Bundle.main.path(forResource: "Res/mockData_baseInterval", ofType: "json"),
              let jsonDataString = try? String(contentsOfFile: jsonPath),
              let canvasView = self.canvasView,
              let chart = self.chart else {
            return
        }
        let jsonData = F2Utils.toJsonArray(jsonDataString)
        chart.canvas()(canvasView).padding()(20, 10, 20, 10)
        chart.source()(jsonData)
        chart.axis()("year", ["grid": false])
        chart.axis()("sales", ["grid": ["type": "dash", "lineWidth": 2, "lineDash": [6, 2]]])
        chart.scale()("year", ["range": [0.1, 0.9]])
        chart.scale()("sales", ["nice": true])
        chart.interval()().position()("year*sales").style()([
            "rounding": [10, 10, 0, 0] // [tl, tr, bl, br]
        ])
        chart.tooltip()([:])
        chart.animate()(true)
        chart.render()()
    }
}
